cub/cub/agent/agent_histogram.cuh

```diff
@@ -630,11 +514,71 @@ struct AgentHistogram
     bool is_valid[PIXELS_PER_THREAD];
 
     // Load tile
-    LoadTile(block_offset, valid_samples, samples, bool_constant_v<IS_FULL_TILE>, bool_constant_v<IS_ALIGNED>);
+    if constexpr (IS_FULL_TILE)
+    {
+      if constexpr (IS_ALIGNED)
+      {
+        if constexpr (NUM_ACTIVE_CHANNELS == 1)
+        {
+          // Load using a wrapped pixel iterator
+          using AliasedPixels = PixelT[PIXELS_PER_THREAD];
+          WrappedPixelIteratorT d_wrapped_pixels((PixelT*) (d_native_samples + block_offset));
+          BlockLoadPixelT(temp_storage.aliasable.pixel_load)
+            .Load(d_wrapped_pixels, reinterpret_cast<AliasedPixels&>(samples));
+        }
+        else
+        {
+          // Load using a wrapped vec iterator
+          using AliasedVecs = VecT[VECS_PER_THREAD];
+          WrappedVecsIteratorT d_wrapped_vecs((VecT*) (d_native_samples + block_offset));
+          BlockLoadVecT(temp_storage.aliasable.vec_load).Load(d_wrapped_vecs, reinterpret_cast<AliasedVecs&>(samples));
+        }
+      }
+      else
+      {
+        using AliasedSamples = SampleT[SAMPLES_PER_THREAD];
+        // Load using sample iterator
+        BlockLoadSampleT(temp_storage.aliasable.sample_load)
+          .Load(d_wrapped_samples + block_offset, reinterpret_cast<AliasedSamples&>(samples));
+      }
+    }
+    else
```

Comment:
Nitpick: Those are nontrivial sections. Could we add
```suggestion
    else // !IS_FULL_TILE
```
Similar to what we do for conditional includes?